server/pbenchinacan/deploy

```diff
@@ -59,24 +60,21 @@ else
 fi
 
 # Update the Nginx favicon and provide some static fallback pages
-podman unshare rm ${SRV_PBENCH}/public_html/favicon.ico
 podman unshare cp ${NGINX_FAVICON} ${SRV_PBENCH}/public_html/favicon.ico
-podman unshare cp /usr/share/nginx/html/404.html /usr/share/nginx/html/50x.html \
-    ${SRV_PBENCH}/public_html/
+podman run --rm --volume ${SRV_PBENCH}:/srv/pbench:Z --entrypoint cp ${PB_SERVER_IMAGE} \
```

Comment:
This is an interesting tidbit of Podman's interface design (which, to be fair, they inherited from Docker).

I believe that the basic idea is that it should be smooth to produce containerized "commands".  So, when you invoke a container, you're not providing an environment in which to run something, rather you're providing an encapsulated _function_ (e.g., a command verb) to which you want to pass arguments.  So, for instance, instead of typing `ls`, you could type `podman run ls_container` (and I'm sure that you can imagine lots of ways to make that phrase easier to type).

The point is, when you invoke a container, the Podman command command line should be focused on providing arguments to the containerized command...and, the command itself shouldn't need to be specified, because it's implicit in the choice of container.

In order to enable this, when the container is built, the entrypoint is set to the intended command.  With this set, the `podman run` command simply passes any trailing arguments on its command line (i.e., after the container name) as arguments to the entrypoint command.  (And, this makes perfect sense, given all the context above.)

The problem crops up in that that hasn't been the way that we on this project have typically used containers -- we've used them as _environments_ in which we run **arbitrary** commands.  And, this is OK -- Podman/Docker supports this model, too:  if the entrypoint for a container is _not_ set, then when the container is invoked, Podman uses the first of the trailing command line arguments as the command to run, and the rest are passed as arguments to the command.  This is the usage model that I think you're referencing above.

However, with this PR, the Pbench Server container now has an entrypoint set.  When the container is normally invoked, we want it to run `/sbin/init`, and so I've set that as the entrypoint, which means that we don't have to specify it on the `podman run` command line.  Note that that also means that, in any _other_ uses of that container (like the interesting one here), we have to explicitly override the entrypoint; otherwise, the invocation above will run `init` and pass an _argument_ of `cp` to it (which I did prove doesn't work out satisfactorily 🥴).

So, what should look weird is
```
podman run <container> <command> <args>
```
 whereas
```
podman run --entrypoint <command> <container> <args>
```
should look normal.  🤷 

So...should I comment something which should be "normal" (and only isn't because of our "atypical" practice)?

Reply:
Fascinating, and makes sense: but it definitely *is* a large departure from our normal, and I guess it's just a learning process.